raiden/utils/echo_node.py

```diff
@@ -96,23 +98,25 @@ def poll_all_received_events(self):
                 if not locked:
                     return
                 else:
-                    received_transfers = self.api.get_channel_events(
+                    received_transfers = self.api.get_payment_history_for_token(
                         self.token_address,
                         from_block=self.last_poll_block,
                     )
+
                     received_transfers = [
-                        event for event in received_transfers
-                        if event['event'] == 'EventPaymentReceivedSuccess'
+                        (block_number, event)
+                        for block_number, event in received_transfers
+                        if type(event) == EventPaymentReceivedSuccess
                     ]
-                    for event in received_transfers:
-                        transfer = event.copy()
-                        transfer.pop('block_number')
+                    for _, event in received_transfers:
+                        transfer = copy.deepcopy(event)
                         self.received_transfers.put(transfer)
+
                     # set last_poll_block after events are enqueued (timeout safe)
                     if received_transfers:
                         self.last_poll_block = max(
-                            event['block_number']
-                            for event in received_transfers
+                            block_number
+                            for block_number, _ in received_transfers
```

Comment:
Maybe I'm misunderstanding the logic here, but the `block_number` and `event` tuple pairs in `received_transfers` only ever seem to be used separately. If that's the case then why store them as tuples in the first place?

Reply:
I thought it would be much easier to understand than this approach: 
```
max = - (2**32 - 1)
                    for block_number, event in received_transfers:
                        if type(event) == EventPaymentReceivedSuccess:
                            # checking for last_pool_block
                            transfer = copy.deepcopy(event)
                            self.received_transfers.put(transfer)
                            if block_number > max:
                                max = block_number
                                self.last_poll_block = max
```

Reply:
>  If that's the case then why store them as tuples in the first place?

That's will be removed in the PR https://github.com/raiden-network/raiden/pull/1859